### PR TITLE
Bump sonata-project/datagrid-bundle to version ^3.0

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because we should bump in this branch. It is potential BC-break.

## Changelog

```markdown
### Added
- support for sonata-project/datagrid-bundle to version ^3.0
### Removed
- support for sonata-project/datagrid-bundle to version < 3.0
```

## To do

- [x] Release `sonata-project/user-bundle`
- [x] Fix `composer.json`
- [x] Bump `knplabs/knp-menu-bundle` to ^3.0 - #769

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
         "sonata-project/notification-bundle": "^3.3",
         "sonata-project/seo-bundle": "^2.4",
         "sonata-project/twig-extensions": "^0.1 || ^1.2",
-        "sonata-project/user-bundle": "^4.0",
+        "sonata-project/user-bundle": "bump_datagrid-dev as 4.4",
```
Comment (VincentLanglet) [inline composer.json:38]: We shouldn't use a fork branch

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (wbloszyk) [reply composer.json:38]: I just add `TODO`.

File: composer.json [hunk 2]
```diff
@@ -35,7 +35,7 @@
         "sonata-project/notification-bundle": "^3.3",
         "sonata-project/seo-bundle": "^2.4",
         "sonata-project/twig-extensions": "^0.1 || ^1.2",
-        "sonata-project/user-bundle": "^4.0",
+        "sonata-project/user-bundle": "4.x-dev as 4.4",
```
Comment (VincentLanglet) [inline composer.json:38]: We shouldn't rely on a dev-branch either.

Do you need a new release of user-bundle now @wbloszyk ? Or maybe we wait for this https://github.com/sonata-project/SonataUserBundle/pull/1237 to be merged too ?

Comment (wbloszyk) [reply composer.json:38]: We should wait for this PR. I should finish checking in tomorrow (In Monday).

**Review by VincentLanglet — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

File: composer.json [hunk 3]
```diff
@@ -36,7 +36,7 @@
         "sonata-project/notification-bundle": "^3.3",
         "sonata-project/seo-bundle": "^2.4",
         "sonata-project/twig-extensions": "^0.1 || ^1.2",
-        "sonata-project/user-bundle": "^4.0",
+        "sonata-project/user-bundle": "^4.9",
```
Comment (OskarStark) [inline composer.json:39]: Is this needed in this PR?

**Review by OskarStark — CHANGES_REQUESTED** [review]
See comment

File: composer.json (hunk 3 above)
Comment (wbloszyk) [reply composer.json:39]: Yes, user-bundle < 4.9 require datagrid 2.x.

**Comment by wbloszyk** [comment]
@OskarStark Can we merge it?
@sonata-project/release-managers Can we release it?

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Thanks

**Pull request merged**